src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -149,7 +163,9 @@ Copyright (c) .NET Foundation. All rights reserved.
     <_SelfContainedWasSpecified Condition="'$(SelfContained)' != ''">true</_SelfContainedWasSpecified>
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(HasRuntimeOutput)' == 'true'">
-    <SelfContained Condition="'$(SelfContained)' == '' and '$(RuntimeIdentifier)' != ''">true</SelfContained>
+    <!-- Breaking change in .NET 8: Projects with 8.0+ TFMS will no longer have RuntimeIdentifier imply SelfContained. Note that PublishReadyToRun will imply SelfContained in these versions. -->
+    <SelfContained Condition="'$(SelfContained)' == '' and '$(RuntimeIdentifier)' != '' and
+                              '$(_TargetFrameworkVersionWithoutV)' &lt; '8.0'">true</SelfContained>
```

Comment:
I dont think we need to check if _TargetFrameworkVersionWithoutV is empty by this point

Reply:
Eh better be safe than sorry

Reply:
I'd suggest switching to `$([MSBuild]::VersionLessThan($(_TargetFrameworkVersionWithoutV), '8.0'))`.

Reply:
Good point, I didnt know about that. Though we need to keep the empty string checks because that function will fail with an empty string